lib/auth/windows/ldap.go

```diff
@@ -180,12 +214,12 @@ func (c *LDAPClient) ReadWithFilter(dn string, filter string, attrs []string) ([
 	)
 	c.mu.Lock()
 	defer c.mu.Unlock()
+
 	res, err := c.client.SearchWithPaging(req, searchPageSize)
-	if ldap.IsErrorWithCode(err, ldap.ErrorNetwork) {
-		return nil, trace.ConnectionProblem(err, "fetching LDAP object %q with filter %q: %v", dn, filter, err)
-	} else if err != nil {
-		return nil, trace.Wrap(err, "fetching LDAP object %q with filter %q: %v", dn, filter, err)
+	if err != nil {
+		return nil, trace.Wrap(convertLDAPError(err), "fetching LDAP object %q with filter %q: %v", dn, filter, err)
```

Comment:
nit: In most cases the error string will be duplicated in our error message: 

`convertLDAPError(err)` returns the full context like: 
`trace.BadParameter("object constraint violation: %v", err)`
and `trace.Wrap(convertLDAPError(err), "... %v", ... err)` will also add full error context.

Reply:
Good catch, thanks!